src/ViewModel/PageLayout/AbstractNestedChildView.php

```diff
@@ -2,56 +2,29 @@
 
 namespace Ingenerator\KohanaView\ViewModel\PageLayout;
 
-use BadMethodCallException;
 use Ingenerator\KohanaView\ViewModel\AbstractViewModel;
 use Ingenerator\KohanaView\ViewModel\NestedChildView;
 use Ingenerator\KohanaView\ViewModel\NestedParentView;
-use Ingenerator\KohanaView\ViewModel\PageLayoutView;
-use UnexpectedValueException;
 
 abstract class AbstractNestedChildView extends AbstractViewModel implements NestedChildView
 {
-    /**
-     * @var NestedParentView
-     */
-    protected $parent_view;
-
-    public function __construct(NestedParentView $parent_view)
-    {
-        $this->parent_view = $parent_view;
+    public function __construct(
+        protected NestedParentView $parent_view,
+    ) {
         parent::__construct();
     }
 
-    /**
-     * @return NestedParentView
-     */
-    public function getParentView()
+    public function getParentView(): NestedParentView
     {
         return $this->parent_view;
     }
 
-    /**
-     * The page layout that this content view will be rendered into.
-     *
-     * @return PageLayoutView
-     */
-    public function var_page()
-    {
-        throw new BadMethodCallException('Call to legacy '.__METHOD__.' interface');
-    }
-
-    /**
-     * @return PageLayoutView
-     */
-    protected function getUltimatePageView()
+    public function getUltimatePageView(): NestedParentView
     {
         $parent = $this->getParentView();
         while ($parent instanceof NestedChildView) {
             $parent = $parent->getParentView();
         }
-        if ( ! $parent instanceof PageLayoutView) {
-            throw new UnexpectedValueException('No ultimate PageLayoutView for '.static::class);
-        }
```

Comment:
This method previously enforced that the top view in the chain had to explicitly be a `PageLayoutView`. This isn't really the responsibility of this class : it's up to the DI container / factories to nest the views as required for the application. The top view can now be any `NestedParentView`.

Also this only worked before because the phpdoc for `->getParentView()` was incorrect: it claimed to always return a `NestedParentView` which would never have passed this instanceof check, but in fact that wasn't the case.